sklearn/utils/estimator_checks.py

```diff
@@ -1309,3 +1310,11 @@ def fit(self, X, y):
 
     assert_true(all(item in deep_params.items() for item in
                     shallow_params.items()))
+
+def check_classifiers_regression_target(name, Estimator):
+    # Check if classifier throws an exception when fed regression targets
+
+    boston = load_boston()
+    X, y_ = boston.data, boston.target
+    e = Estimator()
+    assert_raises(ValueError, e.fit, X, y_)
```

Comment:
Can you check for the error message raised with ``assert_raise_regexp`` or assert_raise_message``?